### ISSUE TITLE
Link Fail JVM_Ge tClassContext Java_java_lang_SecurityManager  19.1.0 ,windows 7 native-image

### ISSUE THREAD

**Issue by lilili87222** [oracle/graal#1478]
java.lib(SecurityManager.obj) : error LNK2019: unresolved external symbol JVM_Ge
tClassContext referenced in function Java_java_lang_SecurityManager_getClassCont
ext
java.lib(SecurityManager.obj) : error LNK2019: unresolved external symbol JVM_Cu
rrentLoadedClass referenced in function Java_java_lang_SecurityManager_currentLo
adedClass0
java.lib(SecurityManager.obj) : error LNK2019: unresolved external symbol JVM_Cu
rrentClassLoader referenced in function Java_java_lang_SecurityManager_currentCl
assLoader0
java.lib(SecurityManager.obj) : error LNK2019: unresolved external symbol JVM_Cl
assDepth referenced in function Java_java_lang_SecurityManager_classDepth
java.lib(SecurityManager.obj) : error LNK2019: unresolved external symbol JVM_Cl
assLoaderDepth referenced in function Java_java_lang_SecurityManager_classLoader
Depth0
G:\demo.exe : fatal error LNK1120: 5 u
nresolved externals

        at com.oracle.svm.hosted.image.NativeBootImageViaCC.write(NativeBootImag
eViaCC.java:343)
        at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator
.java:653)
        at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGe
nerator.java:440)

**Comment by olpaw** [comment]
It seems you are not using the Windows SDK for Windows 7 as described in https://github.com/oracle/graal/blob/master/substratevm/README.md

**Comment by olpaw** [comment]
For reference:
```
Setting SDK environment relative to C:\Program Files\Microsoft SDKs\Windows\v7.1\.
Targeting Windows 7 x64 Debug

C:\Users\pwoegere>cd Downloads

C:\Users\pwoegere\Downloads>cd graalvm-ce-windows-amd64-19.1.0

C:\Users\pwoegere\Downloads\graalvm-ce-windows-amd64-19.1.0\graalvm-ce-19.1.0>notepad Hello.java

C:\Users\pwoegere\Downloads\graalvm-ce-windows-amd64-19.1.0\graalvm-ce-19.1.0>bin\javac.exe Hello.java

C:\Users\pwoegere\Downloads\graalvm-ce-windows-amd64-19.1.0\graalvm-ce-19.1.0>bin\native-image.cmd Hello
[hello:8356]    classlist:   3,203.66 ms
[hello:8356]        (cap):  10,406.00 ms
[hello:8356]        setup:  12,274.79 ms
[hello:8356]   (typeflow):  11,047.13 ms
[hello:8356]    (objects):  11,255.07 ms
[hello:8356]   (features):     942.97 ms
[hello:8356]     analysis:  24,393.46 ms
[hello:8356]     (clinit):     196.33 ms
[hello:8356]     universe:     673.67 ms
[hello:8356]      (parse):   1,310.90 ms
[hello:8356]     (inline):   2,190.43 ms
[hello:8356]    (compile):  13,827.25 ms
[hello:8356]      compile:  17,952.30 ms
[hello:8356]        image:   1,411.35 ms
[hello:8356]        write:   1,349.31 ms
[hello:8356]      [total]:  62,082.49 ms

C:\Users\pwoegere\Downloads\graalvm-ce-windows-amd64-19.1.0\graalvm-ce-19.1.0>hello.exe
Hello Java
```

**Comment by lilili87222** [comment]
Helloword works.  19.0.2 works ,but 19.1.0 not work.
Try use https ,try use KeyStore keystore = KeyStore.getInstance("PKCS12");

**Comment by lilili87222** [comment]
D:\developer\graalvm-ce-19.1.0\bin\native-image --allow-incomplete-classpath --initialize-at-build-time=net.appuploader.logic  --report-unsupported-elements-at-runtime -H:+PrintClassInitialization --no-fallback --enable-url-protocols --enable-all-security-services -H:+ReportExceptionStackTraces  -H:EnableURLProtocols=http -H:ReflectionConfigurationFiles=config/reflect-config.json -H:ResourceConfigurationFiles=G:\config\resource-config.json -jar demo.jar

**Comment by lilili87222** [comment]
Microsoft (R) C/C++ Optimizing Compiler Version 16.00.30319.01 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

cl : Command line warning D9002 : ignoring unknown option '/PDBSTRIPPED'
Microsoft (R) Incremental Linker Version 10.00.30319.01
Copyright (C) Microsoft Corporation.  All rights reserved.

**Comment by olpaw** [comment]
> cl : Command line warning D9002 : ignoring unknown option '/PDBSTRIPPED'

That's just a warning that does not prevent the linking to succeed. I.e Is no related to the issue you are describing.

**Comment by olpaw** [comment]
> Try use https ,try use KeyStore keystore = KeyStore.getInstance("PKCS12");

Please provide a simple reproducer.

**Comment by lilili87222** [comment]
H:EnableURLProtocols=https
Will reproduct

**Comment by lilili87222** [comment]
D:\developer\graalvm-ce-19.1.0\bin\native-image --allow-incomplete-classpath --report-unsupported-elements-at-runtime -H:+PrintClassInitialization --no-fallback --enable-url-protocols --enable-all-security-services -H:+ReportExceptionStackTraces  -H:EnableURLProtocols=https -jar demo.jar

**Comment by lilili87222** [comment]
```
package net.appuploader;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.net.URL;
import java.net.URLConnection;

public class Demo {
	public static void main(String[] args) {
		try {
			doPost("a=a");
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
protected static  String doPost(String param) throws Exception {
		
		OutputStream outStream = null;
		InputStream inputStream = null;
		String result = "";
		try {
			URL realUrl = new URL("https://www.baidu.com");
			URLConnection conn = realUrl.openConnection();
//            conn.setRequestProperty("content-type", "application/json; charset=UTF-8");
			conn.setRequestProperty("accept", "*/*");
			conn.setRequestProperty("connection", "Keep-Alive");
			conn.setRequestProperty("user-agent", "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1;SV1)");
			conn.setDoOutput(true);
			conn.setDoInput(true);
			outStream = conn.getOutputStream();
			outStream.flush();
			StringBuilder builder = new StringBuilder();
			
			result = builder.toString();
		}
		finally {
			try {
				if (outStream != null) {
					outStream.close();
				}
				if (inputStream != null) {
					inputStream.close();
				}
			} catch (IOException ex) {
				ex.printStackTrace();
			}
		}
		return result;

	}
}

```

**Comment by lilili87222** [comment]
security.provider.1=sun.security.provider.Sun
security.provider.2=sun.security.rsa.SunRsaSign
security.provider.3=org.bouncycaste.jce.provider.BouncyCastleProvider
# security.provider.3=sun.security.ec.SunEC
security.provider.4=com.sun.net.ssl.internal.ssl.Provider
security.provider.5=com.sun.crypto.provider.SunJCE
security.provider.6=sun.security.jgss.SunProvider
security.provider.7=com.sun.security.sasl.Provider
security.provider.8=org.jcp.xml.dsig.internal.dom.XMLDSigRI
# security.provider.9=sun.security.smartcardio.SunPCSC
# security.provider.10=sun.security.mscapi.SunMSCAPI

**Comment by lilili87222** [comment]
```
security.provider.1=sun.security.provider.Sun
security.provider.2=sun.security.rsa.SunRsaSign
security.provider.3=org.bouncycaste.jce.provider.BouncyCastleProvider
# security.provider.3=sun.security.ec.SunEC
security.provider.4=com.sun.net.ssl.internal.ssl.Provider
security.provider.5=com.sun.crypto.provider.SunJCE
security.provider.6=sun.security.jgss.SunProvider
security.provider.7=com.sun.security.sasl.Provider
security.provider.8=org.jcp.xml.dsig.internal.dom.XMLDSigRI
# security.provider.9=sun.security.smartcardio.SunPCSC
# security.provider.10=sun.security.mscapi.SunMSCAPI
```

**Comment by olpaw** [comment]
Using your sample and your modifications **on graalvm-ce-19.0.2** gives me:
```
C:\Users\pwoegere\Downloads\graalvm-ce-19.0.2>bin\native-image --allow-incomplete-classpath --report-unsupported-elements-at-runtime -H:+PrintClassInitialization --no-fallback --enable-url-protocols --enable-all-security-services -H:+ReportExceptionStackTraces -H:EnableURLProtocols=https -cp test net.appuploader.Demo
[net.appuploader.demo:9136]    classlist:   4,117.22 ms
[net.appuploader.demo:9136]        (cap):  22,536.48 ms
[net.appuploader.demo:9136]        setup:  24,009.35 ms
[net.appuploader.demo:9136]   (typeflow):  15,351.63 ms
[net.appuploader.demo:9136]    (objects):   7,754.69 ms
[net.appuploader.demo:9136]   (features):     973.81 ms
[net.appuploader.demo:9136]     analysis:  24,758.50 ms
Printing initializer configuration to C:\Users\pwoegere\Downloads\graalvm-ce-19.0.2\reports\initializer_configuration_20190708_115218.txt
Printing initializer dependencies to C:\Users\pwoegere\Downloads\graalvm-ce-19.0.2\reports\initializer_dependencies_20190708_115218.dot
Printing 1 classes that are considered as safe for build-time initialization to C:\Users\pwoegere\Downloads\graalvm-ce-19.0.2\reports\safe_classes_20190708_115218.txt
Printing 3098 classes of type BUILD_TIME to C:\Users\pwoegere\Downloads\graalvm-ce-19.0.2\reports\build_time_classes_20190708_115218.txt
Printing 38 classes of type RERUN to C:\Users\pwoegere\Downloads\graalvm-ce-19.0.2\reports\rerun_classes_20190708_115218.txt
Printing 0 classes of type RUN_TIME to C:\Users\pwoegere\Downloads\graalvm-ce-19.0.2\reports\run_time_classes_20190708_115218.txt
[net.appuploader.demo:9136]     (clinit):     343.77 ms
[net.appuploader.demo:9136]     universe:     770.95 ms
[net.appuploader.demo:9136]      (parse):   1,977.79 ms
[net.appuploader.demo:9136]     (inline):   2,530.52 ms
[net.appuploader.demo:9136]    (compile):  18,287.85 ms
[net.appuploader.demo:9136]      compile:  23,565.09 ms
[net.appuploader.demo:9136]        image:   1,362.53 ms
[net.appuploader.demo:9136]        write:   2,415.86 ms
[net.appuploader.demo:9136]      [total]:  81,522.97 ms
```
Using your sample and your modifications **on graalvm-ce-19.1.0** gives me:
```
C:\Users\pwoegere\Downloads\graalvm-ce-19.0.2>cd ..\graalvm-ce-19.1.0

C:\Users\pwoegere\Downloads\graalvm-ce-19.1.0>bin\native-image --allow-incomplete-classpath --report-unsupported-elements-at-runtime -H:+PrintClassInitialization --no-fallback --enable-url-protocols --enable-all-security-services -H:+ReportExceptionStackTraces -H:EnableURLProtocols=https -cp ..\graalvm-ce-19.0.2\test net.appuploader.Demo
[net.appuploader.demo:2792]    classlist:   3,163.84 ms
[net.appuploader.demo:2792]        (cap):  23,175.02 ms
[net.appuploader.demo:2792]        setup:  25,086.51 ms
[net.appuploader.demo:2792]   (typeflow):  13,989.85 ms
[net.appuploader.demo:2792]    (objects):  10,646.11 ms
[net.appuploader.demo:2792]   (features):     779.30 ms
[net.appuploader.demo:2792]     analysis:  25,792.16 ms
Printing initializer configuration to C:\Users\pwoegere\Downloads\graalvm-ce-19.1.0\reports\initializer_configuration_20190708_115557.txt
Printing initializer dependencies to C:\Users\pwoegere\Downloads\graalvm-ce-19.1.0\reports\initializer_dependencies_20190708_115557.dot
Printing 1 classes that are considered as safe for build-time initialization to C:\Users\pwoegere\Downloads\graalvm-ce-19.1.0\reports\safe_classes_20190708_115557.txt
Printing 3111 classes of type BUILD_TIME to C:\Users\pwoegere\Downloads\graalvm-ce-19.1.0\reports\build_time_classes_20190708_115557.txt
Printing 39 classes of type RERUN to C:\Users\pwoegere\Downloads\graalvm-ce-19.1.0\reports\rerun_classes_20190708_115557.txt
Printing 0 classes of type RUN_TIME to C:\Users\pwoegere\Downloads\graalvm-ce-19.1.0\reports\run_time_classes_20190708_115557.txt
[net.appuploader.demo:2792]     (clinit):     414.66 ms
[net.appuploader.demo:2792]     universe:   1,157.97 ms
[net.appuploader.demo:2792]      (parse):   2,104.33 ms
[net.appuploader.demo:2792]     (inline):   2,765.63 ms
[net.appuploader.demo:2792]    (compile):  18,045.10 ms
[net.appuploader.demo:2792]      compile:  23,664.22 ms
[net.appuploader.demo:2792]        image:   1,584.17 ms
[net.appuploader.demo:2792]        write:   3,370.16 ms
[net.appuploader.demo:2792]      [total]:  84,324.14 ms
```

**Comment by lilili87222** [comment]
[debug_out.zip](https://github.com/oracle/graal/files/3370857/debug_out.zip)


`
..\graalvm-ce-19.1.0\bin\native-image --allow-incomplete-classpath --report-unsupported-elements-at-runtime -H:+PrintClassInitialization --no-fallback --enable-url-protocols --enable-all-security-services -H:+ReportExceptionStackTraces  -H:EnableURLProtocols=http,https -cp swing net.appuploader.Demo
`

I have all log ,java code and trace out in attach.

System  Windows7 64, Chinese or English.
graal 19.1.0

**Comment by lilili87222** [comment]
Addition:
not install mx,not set java home, I download graal 19.1.0 from release page  only

**Comment by lilili87222** [comment]
This error not fix in  19.2 ce

**Comment by olpaw** [comment]
Please reopen if you can reproduce with the latest release.
